pkg/notification/notification.go

```diff
@@ -28,9 +28,9 @@ func (m MessageTemplateParam) calculateExecutionTime() (completionTime *metav1.T
 }
 
 type Notification interface {
-	NotifyStart(messageParam MessageTemplateParam) (err error)
-	NotifySuccess(messageParam MessageTemplateParam) (err error)
-	NotifyFailed(messageParam MessageTemplateParam) (err error)
+	NotifyStart(messageParam MessageTemplateParam, slackChannel string) (err error)
+	NotifySuccess(messageParam MessageTemplateParam, slackChannel string) (err error)
+	NotifyFailed(messageParam MessageTemplateParam, slackChannel string) (err error)
```

Comment:
Since the interface here is implemented in a way that notifications to other platforms can also be implemented.
e.g。 teams, email, github

Is it possible to implement it in a form like without depending on a specific tool?
`NotifyFailed(job v1.Job, messageParam MessageTemplateParam) (err error)`

Reply:
I've refactored a logic to move slack related changes into slack.go.
To achieve this I've added Annotations map[string]string to messageParam, hope this is OK

Reply:
Thank you.
LGTM